### ISSUE TITLE
New playable race: Pixie

### ISSUE THREAD

**Issue by angband-import** [angband/angband#3764]
**Reported by daniel-santos on 3 Nov 2012 19:23 UTC**
OK, this is another experiment in anomalous races.  This one isn't working out too well so far, but I thought that I might share it. (EDIT: I realized after the fact that I enabled "Monsters learn from their mistakes" and had forgotten about it).

```
N:14:Pixie
S:-6:2:4:10:-3:6
R:25:7:15:7:8:12:-20:20:20:-1
X:5:70:3
I:1:10:4
H:21:2:19:2
W:12:1:11:1
F:FEATHER
C:0|1|3|4
```

This is intended to be a highly challenging race, but I find that I generally have to wander dungeons half my level to stay alive.  I think this is a combination of the hit dice of 5 and the low strength.  I've played a pixie ranger to level 19 thus far.  Maybe some of it is just bad luck, not having gotten any artifacts yet.

This would certainly be easier if the hit dice were higher, but to me, that would ruin what I'm going after here. If there was something else that I could add to it (with code changes), it would be centered around allowing them to function more like a mage, where you have to fight from afar and stay out of melee range.

My initial thoughts on this is that they are magical creatures, and therefore, would have the following magical bonuses:
* bonus to spell success, halving the failure chances of any spell cast (any class)
* giving them a chance to use less spell points when they cast a spell, probably based upon their primary spell stat, (i.e., int or wis)
* possibly even enabling the ZERO_FAIL flag for any class (not sure about this one)
* maybe beaming chance when not enabled with class (not sure again)

In addition, there are a few other possibilities that can make this class more workable:
* innate speed bonus, perhaps +1 at level 1, +2 at level 11, +3 at level 21, etc.
* some type of innate survival spells that they can cast every so often (like a built-in rod), things like phase door, teleport, etc.

Obviously, all of the above would be overkill, but two or three I think can make this race work and add a nifty new dimension to the game.

Also, it would need history & an avatar as well.

```
  [Angband v3.5-dev-34-g5a84801-dirty Character Dump]

 Name   Leef                                     Self  RB  CB  EB   Best
 Sex    Male         Age             13   STR:     16  -6  +0  +0     10
 Race   Pixie        Height          22   INT:     17  +2  +2  +0  18/30
 Class  Ranger       Weight          12   WIS:     10  +4  -2  +0     12
 Title  Courser      Social  Well-liked   DEX:     12 +10  +1  +0  18/50
 HP     110/110      Maximize         Y   CON:     16  -3  -1  +0     12
 SP     31/31                             CHR:     10  +6  +1  +1     18

 Level               19   Armor    [21,+42]     Saving Throw         63%
 Cur Exp           3559   Fight     (+3,+0)     Stealth           Superb
 Max Exp           3559   Melee     (+8,+7)     Fighting       Excellent
 Adv Exp           3960   Shoot     (+3,+1)     Shooting          Heroic
 MaxDepth    550' (L11)   Blows    1.6/turn     Disarming            74%
 Game Turns      392960   Shots      1/turn     Magic Device          61
 Standard Turns   35039   Infra       30 ft     Perception       1 in 22
 Resting Turns     3201   Speed          -4     Searching            32%
 Gold              6241   Burden  109.7 lbs

 You are one of several children of a Guildsman.  You are a credit to
 the family.  You have brown eyes, straight black hair, and a dark
 complexion.


rAcid:............. Nexus:.............
rElec:............. Nethr:.............
rFire:............. Chaos:.............
rCold:............. Disen:.............
rPois:............. Feath:............+
rLite:............. pFear:.............
rDark:............. pBlnd:.............
Sound:............. pConf:.............
Shard:............. pStun:.............

Light:.....+....... Tunn.:.............
Regen:............. Speed:.............
  ESP:............. Blows:.............
Invis:............. Shots:.............
FrAct:..........+.. Might:.............
HLife:............. S.Dig:.............
Stea.:............. ImpHP:.............
Sear.:.............  Fear:.............
Infra:............+ Aggrv:.............
```

I'll see what it's like w/o the "Monsters learn from their mistakes" flag enabled.

**Comment by angband-import** [comment]
**Attachment from daniel-santos on 7 Nov 2012 22:14 UTC**

REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_1700_GIT_3764/0001-Add-PF_FAST-player-flag.txt

**Comment by angband-import** [comment]
**Attachment from daniel-santos on 7 Nov 2012 22:14 UTC**

REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_1700_GIT_3764/0002-Add-PF_MAGICAL-player-flag.txt

**Comment by angband-import** [comment]
**Attachment from daniel-santos on 7 Nov 2012 22:14 UTC**

REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_1700_GIT_3764/0003-Make-OF_FEATHER-a-usable-race-flag.txt

**Comment by angband-import** [comment]
**Attachment from daniel-santos on 7 Nov 2012 22:14 UTC**

REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_1700_GIT_3764/0004-Add-Pixie-race.txt

**Comment by angband-import** [comment]
**Attachment from daniel-santos on 18 Nov 2012 09:33 UTC**
updated patch
REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_1700_GIT_3764/0001-Add-PF_FAST-player-flag.txt

**Comment by angband-import** [comment]
**Attachment from daniel-santos on 18 Nov 2012 09:33 UTC**
updated patch
REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_1700_GIT_3764/0002-Add-PF_MAGICAL-player-flag.txt

**Comment by angband-import** [comment]
**Attachment from daniel-santos on 18 Nov 2012 09:33 UTC**
updated patch
REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_1700_GIT_3764/0003-Make-OF_FEATHER-a-usable-race-flag.txt

**Comment by angband-import** [comment]
**Attachment from daniel-santos on 18 Nov 2012 09:34 UTC**
updated patch
REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_1700_GIT_3764/0004-Add-Pixie-race.txt

**Comment by angband-import** [comment]
**Comment by daniel-santos on 7 Nov 2012 22:14 UTC**
OK, so I've played with this some more and added the magical bonuses to mana usage & failure chance as well as the haste one and it did indeed offset the liabilities.  I would be in favor of dropping the hitdie to 4, but I don't really have time right now to play test it again.  Also, I have changed the exp factor to 100, since they don't seem under powered anymore, although they are still darned fragile.

In summary, they have the new player flags PF_FAST and PF_MAGICAL.

PF_HASTE grants a haste bonus of 1 + level * 6 / 50:
```
Level  Bonus
1      +1
9      +2
17     +3
25     +4
34     +5
42     +6
50     +7
```

PF_MAGICAL halves the chance of spell failure (after taking into account all other factors except for priest "icky weapon") and grants chance to cast spells/recite prayers at a reduced cost (10%-60%) or greatly reduced cost (1%-10%) dependent upon level:
* Chance of reduced cost is 5% at level 1 and 50% at level 50, ((lev - 1) * 46 / 50 + 5).
* Chance of greatly reduced cost is 1% at level 10 and 5% at level 50 (lev / 10).

Further adjustment of expfactor may be required, as this now makes them, in some ways, the most powerful race for a Mage class.  As it turns out (I'm still learning this game) the constitution bonus (for a maxed out CON) provides many more hitpoints than the hit die themselves, which will average only 150 at level 50.  The CON bonus is a great deal more.

**Comment by angband-import** [comment]
**Comment by daniel-santos on 18 Nov 2012 10:06 UTC**
Well, tried to replace the patches I submitted earlier with these new ones, but the old ones are still listed :(

So I've played and tweaked more and I'm happy with this now.  I only have a ranger at level 35 and a mage at 27. This is where the Pixie is at now:

```
N:12:Pixie
S:-6:4:4:9:-4:6
R:20:7:15:7:8:12:-20:10:10:-1
X:4:110:3
I:30:10:4
H:21:2:20:2
W:12:1:12:1
F:FEATHER
Y:FAST | MAGICAL
C:0|1|3|4
```

So yes, their hitdie is 4 now and I'm able to survive playing a mage, but it's slow leveling as I'm under level a good deal.  On average, a Pixie Mage will have 75 less hit points at level 50 than a hobbit and 150 less than a human, but of course, constitution makes a big difference.

So I've re-adjusted the haste bonus, it's now 1 + level / 12.5. The haste bonus seems critical for making this race playable at low levels as it helps to offset speed lost from the very low str, especially on a mage!.
```
Level  Speed Bonus
1      +1
13     +2
25     +3
38     +4
50     +5
```
And the magical race bonus reduces spell failure by 1/3rd instead of 1/2 (which seemed too powerful).  Also, the chance to use less manna (i.e., "cast with ease") I've *finally* gotten to what feels right. The chance is now 10%-50% (based upon level), minus the spell's failure chance.  The formula is:
```
10 + (level - 1) * 40 / 49 - fail_chance
```
For example, casting a spell with a 10% failure chance at level 20, would yield a 15% chance to use less manna. This keeps difficult spells from ever being cast "with ease".  Finally, I've adjusted the mana reduction as well, so that the minimum it can be reduced to is also the spell's failure chance and the max is 75% (when the success check is passed, that is).  So in other words, for the same spell as above, if it is cast "with ease", it will use between 10% and 75% of it's normal manna cost.

So to summarize, here is the average manna reduction for casting a spell with a 3% fail rate (not accounting for manna saved by failing less):
```
Level  Avg Reduction
1       4.27%
10      8.54%
20     13.42%
30     18.30%
40     23.18%
50     28.67%
```
I think this makes a nice perk for having a weaker race without making them too powerful.

EDIT: I have also cobbled together a basic history for them, but it too can use some tweaking and refining.

**Comment by angband-import** [comment]
**Comment by takkaria on 5 Apr 2013 04:12 UTC**
Though I appreciate your effort, the bug tracker isn't really the place to introduce big new changes like adding new player races.  I'd suggest posting a patched version on the forums and seeing if people like the idea of adding new races or not.

**Comment by angband-import** [comment]
**Modified by takkaria on 5 Apr 2013 04:26 UTC**

**Comment by angband-import** [comment]
**Comment by daniel-santos on 6 Apr 2013 00:59 UTC**
Sure, I should dig these back up. I like playing games about as much as I enjoy working on them, and I tend to jump around a little.  I did end up with what I thought was a fairly polished troll & pixie mod.  I posted an early rendition of it on the forums, but I never posted a pre-compiled version of it because I don't tend to think that way for some reason.

Anyway, that sounds like a solid plan, but I'll probably get to it in a few weeks as my hands are pretty full at the moment.

**Comment by NickMcConnell** [comment]
Same as for #3757

**Comment by igroglaz** [comment]
:D I just wanted to add pixie race in Tangaria (and I already have troll), so this analysis is very useful. Thanks for bumping it!
